homeassistant/components/radiotherm/__init__.py

```diff
@@ -10,30 +12,47 @@
 from homeassistant.core import HomeAssistant
 from homeassistant.exceptions import ConfigEntryNotReady
 
-from .const import CONF_HOLD_TEMP, DOMAIN
+from .const import DOMAIN
 from .coordinator import RadioThermUpdateCoordinator
-from .data import async_get_init_data
+from .data import RadioThermInitData, async_get_init_data
+from .util import async_set_time
 
-PLATFORMS: list[Platform] = [Platform.CLIMATE]
+PLATFORMS: list[Platform] = [Platform.CLIMATE, Platform.SWITCH]
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Radio Thermostat from a config entry."""
-    host = entry.data[CONF_HOST]
+async def _async_call_or_raise_not_ready(
+    coro: Coroutine[Any, Any, RadioThermInitData | None], host: str
```

Comment:
Maybe we can use a type variable `T` to type the return value of the coroutine so we don't need to use a union since that's not really what the coroutine can return? The coroutine has a return value and this wrapper will always return that return value. It's similar to how we type decorators.

Reply:
Done in b016621bdec9593a6c7330026c5f18af8c6ffabd